library/test/src/lib.rs

```diff
@@ -690,7 +713,7 @@ fn run_test_in_spawned_subprocess(desc: TestDesc, testfn: Box<dyn FnOnce() + Sen
     });
     let record_result2 = record_result.clone();
     panic::set_hook(Box::new(move |info| record_result2(Some(&info))));
-    testfn();
+    testfn().unwrap();
```

Comment:
In general, the error message from this unwrap() is going to be worse than the previous assert_eq error message, I think -- it would be good to have a helper function that handles it via `if let Err` and panics with the String error directly.

Reply:
Done.